scipy/ndimage/morphology.py

```diff
@@ -205,13 +205,10 @@ def generate_binary_structure(rank, connectivity):
     if connectivity < 1:
         connectivity = 1
     if rank < 1:
-        if connectivity < 1:
-            return numpy.array(0, dtype=bool)
-        else:
-            return numpy.array(1, dtype=bool)
+        return numpy.array(1, dtype=bool)
     output = numpy.fabs(numpy.indices([3] * rank) - 1)
     output = numpy.add.reduce(output, 0)
-    return numpy.asarray(output <= connectivity, dtype=bool)
+    return output <= connectivity
```

Comment:
`output` is always an `ndarray`, so a comparison on it will always be a boolean array, the extra call was unnecessary.